introduction-to-datasets.md

```diff
@@ -0,0 +1,119 @@
+    {"name":"imdb_url","type":"string"}
+  ]
+}
+```
+
+The goal is to get the schema into `.avsc` format. The following links provide examples for inferring schemas from data files or Java classes.
+
+| Java API                                                      | Command Line Interface |
+| --------                                                      | ---------------------- |
+| [Inferring a schema from a Java Class][api-schema-from-class] | [Inferring a schema from a Java class][cli-schema-from-class] |
+| [Using the schema of an Avro data file][api-schema-from-data] | [Inferring a schema from a CSV file][cli-schema-from-csv] |
+
+[api-schema-from-class]: {{site.baseurl}}/Inferring-a-Schema-from-a-Java-Class.html
+[api-schema-from-data]: {{site.baseurl}}/Inferring-a-Schema-from-an-Avro-Data-File.html
+[cli-schema-from-class]: {{site.baseurl}}/cli-reference.html#obj-schema
+[cli-schema-from-csv]: {{site.baseurl}}/cli-reference.html#csv-schema
+
+## Partitions
+
+Partitions define logical categories for data storage. For example, you might most often retrieve your data using time-based queries. You can define a partitioning strategy by year, month, day, and hour. When you look for data from January 8, 2015, your search engine only has to look in the data partition `/year=2015/month=1/day=8`. By using partitions that correspond to your most common queries, your data searches run more quickly.
+
+Partitioning is optional, because there are times when partitioning is not the most efficient solution. However, you should always consider partitioning as a best practice when planning your dataset.
+
+When using the CLI, you define your partition strategy in [JSON format][json-format] and apply it when you create your dataset. See [Partitioned Datasets][partition-strategy].
+
+When using the API, you define your dataset in the `DatasetDescriptor.Builder`. See [Partition Builder][partition-builder].
+
+For example, you can use the CLI command `partition-config` to define a partition strategy using name:value pairs to specify that the dataset should be partitioned on a timestamp (`ts`) field by year, month, and day.
+
+```
+user@work:~$ kite-dataset partition-config ts:year ts:month ts:day -s rating.avsc
+```
+The result is a partition definition in JSON format. You can use the `-o` option to out put the result and store it in a .json file.
+
+```JSON
+[ {
+  "name" : "year",
+  "source" : "ts",
+  "type" : "year"
+}, {
+  "name" : "month",
+  "source" : "ts",
+  "type" : "month"
+}, {
+  "name" : "day",
+  "source" : "ts",
+  "type" : "day"
+} ]
+```
+
+[json-format]: {{site.baseurl}}/Partition-Strategy-Format.html
+[partition-strategy]: {{site.baseurl}}/Partitioned-Datasets.html#partition-strategies
+[partition-builder]: {{site.baseurl}}/API-Overview.html#partition-strategy
+
+## Configuration Options
+
+While you need only a URI and schema to create a dataset, you can enhance the performance of your dataset with additional configuration.
```

Comment:
I just realized this is to introduce the column mapping and properties sections, so my comment above isn't quite accurate. The note about enhancing the performance isn't correct. As I noted above, properties do more than just performance and are intended to expose some control over how Kite behaves while reading or writing. Column mappings affect performance, but the real take-away is that column mappings are required to store data in HBase.